pallets/nfts/src/tests.rs

```diff
@@ -1935,60 +2323,314 @@ fn approving_multiple_accounts_works() {
+			item_owner.clone(),
+			default_item_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner.clone()),
+			locked_collection,
+			item,
+			item_owner.clone(),
+			default_item_config()
 		));
-		assert_eq!(
-			approvals(0, 42),
-			vec![(account(3), None), (account(4), None), (account(5), Some(current_block + 2))]
+
+		// Throws error `Error::ItemsNonTransferable`.
+		assert_ok!(Nfts::lock_collection(
+			RuntimeOrigin::signed(collection_owner),
+			locked_collection,
+			CollectionSettings::from_disabled(CollectionSetting::TransferableItems.into())
+		));
+		assert_noop!(
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				locked_collection,
+				account(3),
+				None
+			),
+			Error::<Test>::ItemsNonTransferable
 		);
 
-		assert_ok!(Nfts::transfer(RuntimeOrigin::signed(account(4)), 0, 42, account(6)));
+		// Approve unknown collection, throws error `Error::NoItemOwned`.
 		assert_noop!(
-			Nfts::transfer(RuntimeOrigin::signed(account(3)), 0, 42, account(7)),
-			Error::<Test>::NoPermission
+			Nfts::approve_collection_transfer(
+				RuntimeOrigin::signed(item_owner.clone()),
+				2,
+				delegate.clone(),
+				None
+			),
+			Error::<Test>::NoItemOwned
 		);
+
+		// Approval expires after `deadline`.
+		let deadline = 10;
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(item_owner.clone()),
+			collection,
+			delegate.clone(),
+			Some(deadline)
+		));
+		assert_ok!(Nfts::check_approval(&collection, &None, &item_owner, &delegate));
+		System::set_block_number(deadline + 2);
 		assert_noop!(
-			Nfts::transfer(RuntimeOrigin::signed(account(5)), 0, 42, account(8)),
-			Error::<Test>::NoPermission
+			Nfts::check_approval(&collection, &None, &item_owner, &delegate),
+			Error::<Test>::ApprovalExpired
 		);
+
+		// Approve delegate of an existing expired approval to transfer.
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(item_owner.clone()),
+			collection,
+			delegate.clone(),
+			Some(deadline)
+		));
+		assert_eq!(Balances::reserved_balance(&account(2)), 1);
+		let now = System::block_number();
+		assert!(events().contains(&Event::<Test>::TransferApproved {
+			collection: 0,
+			item: None,
+			owner: item_owner.clone(),
+			delegate: delegate.clone(),
+			deadline: Some(now + deadline)
+		}));
+		assert_eq!(
+			CollectionApprovals::<Test>::get((0, item_owner.clone(), delegate.clone())),
+			Some((Some(now + deadline), 1))
+		);
+
+		// Approve same delegate again not updating the total reserved funds.
+		assert_ok!(Nfts::approve_collection_transfer(
+			RuntimeOrigin::signed(item_owner.clone()),
+			collection,
+			delegate.clone(),
+			None
+		));
+		assert_eq!(Balances::reserved_balance(&item_owner), 1);
+
+		assert_ok!(Nfts::transfer(RuntimeOrigin::signed(delegate), collection, item, account(4)));
 	});
 }
 
 #[test]
-fn approvals_limit_works() {
+fn force_approve_collection_transfer_works() {
```

Comment:
This code can combine with the `approve_collection_transfer_works` test to reduce code duplication by having a for loop over the `[(Runtime::signed(origin), None), (Runtime::root(), Some(owner))]` but it need additional helper method. I'm not sure if it is better as keeping them in a separate test case keep it simple, but increase maintenance effort.

Reply:
I take my words back we can easily reduce the tests by using a simple for loop: `for origin in [ForceOrigin, owner]`

Reply:
Resolved in the next commit by combining the `force_` test with the another together.